lib/MusicBrainz/Server/Controller/Role/WikipediaExtract.pm

```diff
@@ -6,23 +6,18 @@ use namespace::autoclean;
 after show => sub {
     my ($self, $c) = @_;
 
-    $self->_get_extract($c, 1);
+    $c->stash->{wikipedia_extract} = $self->_get_extract($c, 1);
 };
 
 sub wikipedia_extract : Chained('load') PathPart('wikipedia-extract')
 {
     my ($self, $c) = @_;
 
-    $self->_get_extract($c, 0);
+    my $wp_extract = $self->_get_extract($c, 0);
 
     $c->res->headers->header('X-Robots-Tag' => 'noindex');
-    $c->stash(
-        current_view => 'Node',
-        component_path => 'components/WikipediaExtract',
-        component_props => {
-            wikipediaExtract => $c->stash->{wikipedia_extract},
-        }
-    );
+    $c->res->content_type('application/json; charset=utf-8');
+    $c->res->{body} = $c->json_utf8->encode({wikipediaExtract => $wp_extract});
```

Comment:
I changed the WP extract component to work in the same way the commons image one does. It returns JSON which can be used as props by React to render the component, rather than returning pre-rendered HTML which has to be spliced into the page with jQuery.